### ISSUE TITLE
It adds black background to the image when converting from PNG to JPEG and resizing

### ISSUE THREAD

**Issue by paulandroshchuk** [Intervention/image#1251]
**Describe the bug**
Mere PNG to JPEG works fine. It converts it and the background is just white.

However, when I'm trying to convert a PNG to JPEG **_and_** resize it using scaleDown, it suddenly adds black background to the image.

* Notice that it adds black background only when I combine PNG to JPEG and resizing.

**Code Example**
```php
ImageManager::gd()
    ->read($pathToPng)
    ->modify(new class implements ModifierInterface {
        public function apply(ImageInterface $image): ImageInterface
        {
            // This line below adds black background to the image.
            // If you comment it out, it will work fine.
            $image->scaleDown(1400, 1400);

            return $image;
        }
    })
    ->toJpeg();
```

**Expected behavior**
Background stays the same.

**Images**
<details>
<summary>Original PNG Image</summary>

![dog-png](https://github.com/Intervention/image/assets/13240539/68cd740a-40b4-4ba6-a118-a71df5fdffb8)

</details>

<details>
<summary>Image when converted to JPEG</summary>

![a1808515-548e-4b6d-986c-57bcf15fb35d](https://github.com/Intervention/image/assets/13240539/aa119743-b7a6-47f4-92be-daf559d10bcf)

</details>

<details>
<summary>Image when converted to JPEG *and* resized (black background)</summary>

![e3c5d014-f2d0-407b-8205-838f19d92c8b](https://github.com/Intervention/image/assets/13240539/0bb16db1-f590-4ce3-9bdf-3d5397368ab3)

</details>

**Environment (please complete the following information):**
 - PHP Version: **8.2.3**
 - OS: **macOS 14.0**
 - Intervention Image Version: **3.2.1**
 - GD or Imagick: GD

**Comment by olivervogel** [comment]
Thanks for reporting. The issue described also happens with Imagick. I will have a look at it.

**Comment by inndim** [comment]
This problem also occurs when using $image->cover(640, 400, 'center'); adds black background when I save png as jpg

**Comment by olivervogel** [comment]
When transforming PNG files with transparent areas into formats that do not support transparency, the transparency is replaced by a color.

This color is white by default, but can be adjusted by specifying it yourself.

The feature will be available as of version 3.3 with more detailed documentation on [intervention.io](https://image.intervention.io/v3).

```php
use Intervention\Image\ImageManager;
use Intervention\Image\Drivers\Gd\Driver;

// create new manager instance with desired driver
$manager = new ImageManager(Driver::class);

// read a PNG file with transparency
$image = $manager->read('images/example.png');

// define the replacement color for transparent areas
$image->setBlendingColor('#ff5500'); // orange

// encode file as jpeg
$image->toJpeg();
```
